.github/workflows/release.yml

```diff
@@ -78,19 +114,51 @@ jobs:
           ref: master
           token: ${{ secrets.NOIR_REPO_TOKEN }}
           inputs: '{ "noir-ref": "${{ needs.release-please.outputs.tag-name }}", "npm-tag": "latest" }'
-  
 
   publish-docs:
     name: Publish docs
     needs: [release-please]
     if: ${{ needs.release-please.outputs.tag-name }}
     runs-on: ubuntu-latest
     steps:
-      - name: Dispatch to publish workflow
-        uses: benc-uk/workflow-dispatch@v1
+      - name: Checkout release branch
+        uses: actions/checkout@v4
         with:
-          workflow: docs-new-version.yml
-          repo: noir-lang/noir
-          ref: master
-          token: ${{ secrets.GITHUB_TOKEN }}
-          inputs: '{ "tag": "${{ needs.release-please.outputs.tag-name }}"}'
+          ref: ${{ needs.release-please.outputs.tag-name }}
+          token: ${{ secrets.NOIR_RELEASES_TOKEN }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '18'
+
+      - name: Install wasm-bindgen-cli
+        uses: taiki-e/install-action@v2
+        with:
+          tool: wasm-bindgen-cli@0.2.86
+
+      - name: Install wasm-opt
+        run: |
+          npm i wasm-opt -g
+      
+      - name: Install Yarn dependencies
+        uses: ./.github/actions/setup
+
+      - name: Build docs for deploying
+        working-directory: docs
+        run: 
+          yarn workspaces foreach -Rt run build
+
+      - name: Deploy to Netlify
+        uses: nwtgck/actions-netlify@v2.1
+        with:
+          publish-dir: './docs/build'
+          production-branch: master
+          production-deploy: true
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          enable-github-deployment: false
+          deploy-message: "Deploy from GitHub Actions for tag ${{ needs.release-please.outputs.tag-name }}"
+        env:
+          NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+          NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
+        timeout-minutes: 1
```

Comment:
If we are keeping pre-release vs release, we should avoid publishing docs on pre-releases to minimize user confusion.

Reply:
On why we need (c):

Tackling (a) with [on: release](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release) would bind the action to the pre-release commit, rather than the latest master at when the pre-release is flipped into a stable release.

Assuming (b) and (c) works on latest master instead, (c) is helpful for scenarios where:

We pre-released a version
We add changes to its versioned docs
We flip the version into a stable release
Docs automatically published from (1) but without changes in (2)
We use (c) to patch the published docs

Reply:
The `build` script already filters for the stable releases already.

A quick explainer on how versioning works:
- the `versions.json` file is the source of truth. If it's there, it will be built and published.
- the `versioned_docs` and `versioned_sidebars` folders are just the assets
- cutting versions simply copies whatever is in `docs` and `sidebars` and pastes into `versioned_docs` and `versioned_sidebars`

The build script simply gets all the releases from GitHub, filters out the non-stable ones, and writes the last 3 stables to the `versions.json`.

So we can cut a version for every prerelease. We have to, because we don't know in advance if a prerelease will be or not be stable later on.